api/server/server.go

```diff
@@ -260,12 +264,74 @@ func NewServer(conf Config) (*Server, error) {
 	}
 
 	s.indexServer = startIndexHTTPServer(s)
+	if err != nil {
+		return nil, fmt.Errorf("starting index server: %s", err)
+	}
+
+	s.ipfsRevProxy, err = startIPFSRevProxy(&conf, ffsManager)
+	if err != nil {
+		return nil, fmt.Errorf("starting IPFS reverse proxy: %s", err)
+	}
 
 	log.Info("Starting finished, serving requests")
 
 	return s, nil
 }
 
+type ffsHTTPAuth struct {
+	cont       http.Handler
+	ffsManager *manager.Manager
+}
+
+func newHTTPFFSAuthInterceptor(cont http.Handler, m *manager.Manager) *ffsHTTPAuth {
+	fha := &ffsHTTPAuth{
+		cont:       cont,
+		ffsManager: m,
+	}
+
+	return fha
+}
+
+func (fha *ffsHTTPAuth) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
```

Comment:
`ffsHTTPAuth` is an "interceptor" which verifies that all request passing through the IPFS reverse proxy are authenticated with an FFS token.